File: bit_manipulation/power_of_two/C/power_of_two.c
#include <stdio.h>
#include <stdbool.h>

bool isPowerOfTwo (int x) {
    /**
     * Function to check if a given number is a power of 2 
     *      x              : checks if x == 0 and 
     *      !(x & (x - 1)) : checks if x is a power of 2 
     */
    return (x && !(x & (x - 1)));
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    
    if (isPowerOfTwo(num)) 
        printf("Given number is a power of 2\n");
    else 
        printf("Given number is not a power of 2\n");

    return 0;
}

File: bit_manipulation/power_of_two/python/power_of_two.py
# Check if given number is power of 2 or not  
  
# Function to check if x is power of 2 
def isPowerOfTwo (x): 
  
    # First x in the below expression is for the case when x is 0  
    return (x and (not(x & (x - 1))) ) 
  
# Driver code 
x = int(input("Enter a no:"))
if(isPowerOfTwo(x)): 
    print('Yes') 
else: 
    print('No') 
